### DISCUSSION TITLE
Support or implement ivi-html

### DISCUSSION THREAD

**Discussion by igotfr** [solidjs/solid#1578] [Ideas]
Support or implement ivi-html as an alternative to hyperscript

https://github.com/localvoid/ivi#quick-start

**Comment by lxsmnsyc** [comment]
solid has hyperscript and html tagged templates supported via `solid-js/h` and `solid-js/html`, respectively. However, neither supports SSR.

**Reply by igotfr** [reply]
ivi-html can be an abstraction on hyperscript, as JSX sucks, it doesn't need to be something new

**Reply by lxsmnsyc** [reply]
@cindRoberta The requirement Solid needs is fine-grained updates. ivi-html doesn't support it, and ivi-html uses Virtual DOM, which Solid doesn't use, so eventually this whole thing isn't just going to be compatible for Solid.

Also, I don't really get what value this brings over `solid-js/h`

**Reply by igotfr** [reply]
you didn't understand, I want only the syntax of ivi-html (or similar), not the VDOM

````ts
div('#id .class', {attrs: attrs_values}, childs);

// instead of
h('div #id .class', {attrs: attrs_values}, childs);
```

**Reply by atk** [reply]
What stops you from providing that abstraction yourself?

```js
function createTag(name) {
  return Object.assign(
    (...args) => h(...(
      typeof args[0] === 'string'
      ? [`${name} ${args[0]}`, ...args.slice(1)]
      : [name, ...args]
    )),
    { name }
  );
}
```

Or something like this.
